frontend/src/app/page.tsx

```diff
@@ -1,20 +1,109 @@
+import { ByDateResponse, calendarClient } from './api/calendarClient';
 
-import dynamic from 'next/dynamic';
-import { ToastContainer } from 'material-react-toastify';
-import 'react-toastify/dist/ReactToastify.css';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+} from '@mui/material';
+import EventCard from './components/EventCard';
 
-const MyTimePicker = dynamic(() => import('@/app/components/CreateEventModal'), {
-  ssr: false,
-});
+const weekdays = [
+  'Monday',
+  'Tuesday',
+  'Wednesday',
+  'Thursday',
+  'Friday',
+  'Saturday',
+  'Sunday',
+];
 
+const cellStyle = {
+  fontWeight: 'bold',
+  color: 'white',
+  backgroundColor: '#1976d2',
+  borderRight: '1px solid white',
+};
+
+const EventsTable = () => {
+  const [events, setEvents] = useState<ByDateResponse>({});
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  async function loadData() {
+    const monday: Date = new Date();
+    monday.setDate(monday.getDate() - monday.getDay() + 1);
+
+    const sunday: Date = new Date();
+    sunday.setDate(sunday.getDate() - sunday.getDay() + 7);
+
+    const eventTest = calendarClient.displayEvents(
+      '00000300-0000-0000-0000-000000000000',
+      monday,
+      sunday
+    );
+    setEvents(await eventTest);
+  }
 
-export default function Page() {
   return (
-    <div>
-      <MyTimePicker />
-      <ToastContainer />      
-    </div>
+    <TableContainer
+      component={Paper}
+      sx={{ maxWidth: 1600, margin: 'auto', borderRadius: 3, mt: 4 }}>
+      <Table>
+        <TableHead>
+          <TableRow>
+            {weekdays.map((day, index) => (
+              <TableCell
+                key={index}
+                align='center'
+                style={{
+                  width: '14.28%',
+                  verticalAlign: 'top',
+                  ...cellStyle,
+                  borderRight:
+                    index === weekdays.length - 1
+                      ? 'none'
+                      : cellStyle.borderRight,
+                }}>
+                {day}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          <TableRow>
+            {weekdays.map((dayName, index) => (
+              <TableCell
+                key={index}
+                align='center'
+                style={{
+                  width: '14.28%',
+                  verticalAlign: 'top',
+                  padding: '8px',
+                  borderRight:
+                    index === weekdays.length - 1 ? 'none' : '1px solid #ccc',
+                }}>
+                {Object.keys(events)
+                  .filter(date => (index + 1) % 7 == new Date(date).getDay())
+                  .map(date => events[date])
+                  .flat()
+                  .map((event, eventIndex) => (
+                    <EventCard key={eventIndex} event={event} />
+                  ))}
```

Comment:
This is not great. Creating a separate component for the day-cell that handles this logic outside the render loop would be better. After that, the array operations can be simplified. In this case, there shouldn't need to be more than a single .map operation in a given array.